### DISCUSSION TITLE
Relationship Column values taking over 22 seconds to get values

### DISCUSSION THREAD

**Discussion by giacomouberti** [Budibase/budibase#8136] [Help]
Hi there, I have an issue with the time my application needs to **load the values from a relationship column**

I'm running a **budibase droplet on DigitalOcean** with an external Postgre managed database also on DigitalOcean but I'm testing the same app **also on my local machine using Docker** and there is no significative difference, so I would exclude a networking or machine issue.

I have a table containing the offer rows where I use two relationship columns to link to the products and brands tables.
My problem is that when I try to edit one of these rows from a screen using a form it takes over 22 seconds to get the list of values to be shown in the dropdown fields.

<img width="1058" alt="Schermata 2022-10-05 alle 13 21 25" src="https://user-images.githubusercontent.com/34889625/194049651-8e509f0d-d11e-464b-bfbe-dccb73d5bfbc.png">

I suspect one of the reasons contributing to the increase in the time necessary to be able to render a usable form is influenced by the fact that the response contains not only the list of values that could be selected but the JSON Payload contains **all the fields of each row of the related table**.

Am I doing something wrong in setting up the relationships and/or the form? Unfortunately, this situation is causing a real issue for the user that needs to wait almost 30 seconds each time they need to update or create a row.

**Comment by melohagan** [comment]
Hey @giacomouberti 

This seems quite similar to: https://github.com/Budibase/budibase/discussions/5299

Luckily in your case you are using Postgres. This means you can use a [custom query](https://docs.budibase.com/docs/custom-queries) to do all the joins, etc and retrieve only the data you need.
